odl/test/discr/lp_discr_test.py

```diff
@@ -754,6 +754,15 @@ def test_astype():
     assert cdiscr.astype('float64') == rdiscr
     assert cdiscr.real_space == rdiscr
 
+    # More exotic dtype
+    discr = odl.uniform_discr([0, 0], [1, 1], [2, 2], dtype=bool)
+    as_float = discr.astype(float)
+    assert as_float.dtype == float
+    assert not as_float.is_weighted
+    as_complex = discr.astype(complex)
+    assert as_complex.dtype == complex
+    assert not as_complex.is_weighted
+
```

Comment:
Do we have any "raises" test that validates that it actually gives `AttributeError`? It seems not?

Reply:
There's a test for that in `fspace_test.py`.

Reply:
Shouldn't it need to be updated then?

Reply:
I see it has been :)

Reply:
Yep!